File: fixtures/cpp/Duck.cpp
#include "Duck.h"
#include <iostream>

Duck::Duck(const std::string& name, int age, const std::string& featherColor)
    : Animal(name, age), featherColor(featherColor) {}

Duck::~Duck() {}

void Duck::fly() {
    std::cout << name << " is flying with " << featherColor << " feathers" << std::endl;
}

void Duck::land() {
    std::cout << name << " is landing" << std::endl;
}

void Duck::swim() {
    std::cout << name << " is swimming" << std::endl;
}

void Duck::makeSound() const {
    std::cout << name << " says: Quack! Quack!" << std::endl;
}

std::string Duck::getFeatherColor() const {
    return featherColor;
}

File: fixtures/java/Duck.java
package com.example.model;

import com.example.interfaces.Flyable;
import com.example.interfaces.Swimmable;

/**
 * 鸭子类 - 同时实现多个接口
 */
public class Duck extends Animal implements Flyable, Swimmable {
    private String featherColor;
    
    public Duck(String name, int age, String featherColor) {
        super(name, age);
        this.featherColor = featherColor;
    }
    
    @Override
    public void fly() {
        System.out.println(name + " is flying with " + featherColor + " feathers");
    }
    
    @Override
    public void land() {
        System.out.println(name + " is landing");
    }
    
    @Override
    public void swim() {
        System.out.println(name + " is swimming");
    }
    
    @Override
    public void makeSound() {
        System.out.println(name + " says: Quack! Quack!");
    }
    
    public String getFeatherColor() {
        return featherColor;
    }
}
